packages/editor/src/components/post-visibility/label.js

```diff
@@ -24,8 +24,11 @@ export default function PostVisibilityLabel() {
  * @return {string} Post visibility label.
  */
 export function usePostVisibilityLabel() {
-	const visibility = useSelect( ( select ) =>
-		select( editorStore ).getEditedPostVisibility()
+	const visibility = useSelect(
+		( select ) => select( editorStore ).getEditedPostVisibility(),
+		[]
 	);
-	return visibilityOptions[ visibility ]?.label;
+
+	return VISIBILITY_OPTIONS.find( ( option ) => option.value === visibility )
+		?.label;
```

Comment:
Just a question: is `find()` more expensive than the previous implementation that was simpler because the options object had keys?

Reply:
Had to change `visibilityOptions` from an object into an array for `RadioControl`. There should be no noticeable difference here.